### PR TITLE
Use .decode()

### LINKED ISSUE #9: Use the image.decode() api
image.decode is part of the new spec, which decodes images and returns a promise, see if this can be incorporated into it. Also try to check for browser support. And have a silent fallback / failing mechanism

### PR DESCRIPTION
**Pull request by kurtextrem**
Fixes #9

### CODE REVIEW

File: lib/ImageWorker.js [hunk 1]
```diff
@@ -79,8 +79,8 @@ class ImageWorker extends Component<ImageWorkerProps, ImageWorkerState> {
     const image = new Image();
     this.image = image;
     
-    image.decode !== undefined ? image.decode().then(this.onLoad) : image.onload = this.onLoad;
     image.src = url;
+    image.decode !== undefined ? image.decode().then(this.onLoad).catch(this.onLoad) : image.onload = this.onLoad;
```
Comment (nitish24p) [inline lib/ImageWorker.js]: !== undefined isnt needed. Just image.decode will suffice

**Review by nitish24p — CHANGES_REQUESTED** [review]

File: lib/ImageWorker.js (hunk 1 above)
Comment (kurtextrem) [reply lib/ImageWorker.js]: The explicit check gives higher performance, see here: 
c7oofqixqaariog

Comment (nitish24p) [reply lib/ImageWorker.js]: Really.. interesting.. thanks. Learnt something new today..

**Review by nitish24p — APPROVED** [review]

**Comment by nitish24p** [comment]
@manjula91 looks mergable..

File: lib/ImageWorker.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ const webWorkerScript = `
         cache: 'default'
     }).then(response => {
         return response.blob();
-    }).then(_ => postMessage(url));
+    }).then(_ => postMessage(url)).catch(console.error);
```
Comment (nitish24p) [inline lib/ImageWorker.js:14]: Just a thought.. do we need to log the error..or use postmessage in the failure block, and load the image normally? without the worker fetch

Comment (kurtextrem) [reply lib/ImageWorker.js:14]: That's a good idea, although most of the time this error means 404 etc, so the image won't be loaded anyway.

Comment (manjula-dube) [reply lib/ImageWorker.js:14]: we should log when its an error :)

Comment (manjula-dube) [reply lib/ImageWorker.js:14]: I think we are good to go :)

@kurtextrem good job :)

File: lib/ImageWorker.js (hunk 1 above)
Comment (manjula-dube) [reply lib/ImageWorker.js]: I dint know about this. Nice one

**Comment by manjula-dube** [comment]
awesome @kurtextrem 👍

**Comment by kurtextrem** [comment]
🎉

**Pull request merged**